kiwi/markup/base.py

```diff
@@ -55,6 +58,12 @@ def apply_xslt_stylesheets(self, description: str) -> str:
         try:
             parsed_description = etree.parse(description)
         except etree.XMLSyntaxError:
+            with open(description) as image_description:
+                content = image_description.readlines()
+                if content[0].startswith('<?xml '):
```

Comment:
I see several issues with this "XML input type detection". Please also note we had a relatively long conversation with Fergel at the time when we added the message for this exception. Without anymarkup in place it's not easy to detect the markup type of the given input and we actually wanted to use the filetype magic for the detection. But also the filetype magic has potential to fail.

The way you solved this unfortunately exposes similar problems. Here are the ones I have tested

1. ```<?xml>``` tag is an optional tag and not mandatory for XML input. It's also written that way in the XML documentation. Thus if you simply drop that line it can still be valid or invalid XML or anything else and your check has no opportunity to check that

2. The expectation that the first line is the one with the ```<?xml>``` tag makes this less robust. The header information must be first prior any markup data but there can be spaces/emptylines or any other sort of non XML markup related characters in the input stream. In your case just adding an empty line before the tag breaks the check. I think that part can be fixed easily

I think what is being tried here, to know for sure that the intention of the user was to validate against XML, cannot be solved in a generic way. This is also the reason why we did not add more or less sophisticated ways to try the guessing game.

However, I agree with you that the error message is misleading if the XML input is invalid XML and expected to be XML. Wouldn't it be an acceptable change to just give the error message more scope ?

Something like

```python
message = dedent('''\n
    The configuration file was validated against the XML markup and failed
    with a syntax error. Please rerun kiwi with --debug to get further details
    on the error if your configuration is intended to be in XML. In case your
    configuration is using another markup language please note that this
    requires the installation of the Python anymarkup module.
''')
try:
    parsed_description = etree.parse(description)
except etree.XMLSyntaxError as issue:
    log.debug(issue)
    raise KiwiConfigFileFormatNotSupported(message)
```

If you would consider this as an option we only need to change the code to print the etree.parse syntax error message from the exception as a log.debug message.

From my perspective this is a good compromise to the user friendly aspect and also not misleading

Thoughts ?

Reply:
I'm sure you can also come up with a much better message but I hope you get the idea ?